alphabase/smiles/smiles.py

```diff
@@ -0,0 +1,99 @@
+from typing import Optional
+import os
+from rdkit import Chem
+from rdkit.Chem.rdmolops import ReplaceSubstructs, SanitizeMol
+
+from alphabase.constants._const import CONST_FILE_FOLDER
+from alphabase.yaml_utils import load_yaml
+
+SMILES_ALL: dict = load_yaml(os.path.join(CONST_FILE_FOLDER, "smiles.yaml"))
+
+N_TERM_PLACEHOLDER = "[Xe]"
+C_TERM_PLACEHOLDER = "[Rn]"
+
+aa_smiles = SMILES_ALL["aa_smiles"]
+n_term_modifications = SMILES_ALL["n_term_modifications"]
+c_term_modifications = SMILES_ALL["c_term_modifications"]
+ptm_dict = SMILES_ALL["ptm_dict"]
+
+aa_mols = {aa: Chem.MolFromSmiles(smile) for aa, smile in aa_smiles.items()}
+
+
+def validate_correct_args(
+    aa_smiles: str, n_term_mod: Optional[str] = None, c_term_mod: Optional[str] = None
+) -> None:
+    """
+    Validate the amino acid, N-terminal modification, and C-terminal modification.
+
+    Args:
+        aa_smiles (str): SMILES of an amino acid with or without PTMs, has to have the placeholder atoms [Xe] for N-term and [Rn] for C-term.
+        n_term_mod (Optional[str]): N-terminal modification name. Options are the keys in the n_term_modifications dict.
+        c_term_mod (Optional[str]): C-terminal modification name. Options are the keys in the c_term_modifications dict.
+
+    Raises:
+        ValueError: If the amino acid is invalid, or if N-terminal / C-terminal modification is not recognized.
+    """
+    if N_TERM_PLACEHOLDER not in aa_smiles or C_TERM_PLACEHOLDER not in aa_smiles:
```

Comment:
could it be that multiple of these errors happen at once? if so, consider this pattern:
```
errors =[]
if N_TERM_PLACEHOLDER not in aa_smiles or C_TERM_PLACEHOLDER not in aa_smiles:
  errors.append(f"The amino acid {aa_smiles} must contain the N-terminal and C-terminal placeholders in order to be modified.")
if ...
if ....

if errors:
  ValueError("\n".join(errors))

```

This way, the user gets all errors at once, and doesnt have to play "error ping pong" :-)

Reply:
Fixed :)